### DISCUSSION TITLE
Not able to set filtermodeoptions for a column with server side paging and filtering

### DISCUSSION THREAD

**Discussion by SRGajjala** [KevinVandy/material-react-table#35] [General]
I would like to set column filter mode options to the column definition mentioned below with server side paging and filtering but not able to see specified columnmodeoptions.
I am seeing all columnmodeoptions.
{
            id: 'gender',
            header: 'Gender',
            enableClickToCopy:true,
            columnFilterModeOptions: ['fuzzy','contains', 'startsWith', 'endsWith','equals','notEquals','empty','notEmpty'],
          }
one more issue with changing of columnfiltermode, suppose if we select **equals** column filter mode with input value 7, after that if you change different filter mode **Not Equals**, it is not taking input value 7 or it is not clearing input value.

**Comment by KevinVandy** [comment]
hmm, all of the examples on the docs site are using SSR, and are working fine. I'll have to dig into it.

I'm aware of the last issue.

**Reply by SRGajjala** [reply]
Use the github source code of https://www.material-react-table.com/docs/examples/remote, just add **columnFilterModeOptions**: ['fuzzy', 'contains', 'startsWith'], to any column definition and add attribute **enableColumnFilterChangeMode** to    **MaterialReactTable** react element. You can find the issue.

When you remove the manualFiltering on MaterialReactTable react element, then specific mentioned columnfiltermodeoptions is showing.

**Reply by SRGajjala** [reply]
I am able to resolve by updating with latest version. 
Clearing the input value whenever the user changes filter mode option is still not solved. Can you please look in to this?
